connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -376,8 +376,13 @@ object SparkConnectService extends Logging {
     uiTab.foreach(_.detach())
   }
 
-  def extractErrorMessage(st: Throwable): String = {
-    val message = StringUtils.abbreviate(st.getMessage, 2048)
+  def extractErrorMessage(st: Throwable, format: ErrorMessageFormat.Value): String = {
+    val message = st match {
+      case e: SparkThrowable =>
+        SparkThrowableHelper.getMessage(e, format)
```

Comment:
Should we restrict the message size by 2048 as well? Some error formats might include full stack traces, and you can face to some resource restrictions like message, buffer sizes while building proto back.

Reply:
thanks i made an update. my assumption is STANDARD/MINIMAL will be able to be within the 2048 limit. otherwise i propose we increase the limit since that would make the format unusuable

Reply:
Some legacy error formats still includes full stack traces (even nested). The stack traces can be easily more than 2048 bytes.

Reply:
oh darn i see what you mean with [this line](https://github.com/apache/spark/blob/01bb1b1a3dbfc68f41d9b13de863d26d587c7e2f/common/utils/src/main/scala/org/apache/spark/SparkThrowableHelper.scala#L87)